chalk-solve/src/clauses/builtin_traits/fn_family.rs

```diff
@@ -88,7 +88,10 @@ pub fn add_fn_trait_program_clauses<I: Interner>(
     self_ty: Ty<I>,
 ) -> Result<(), Floundered> {
     let interner = db.interner();
-    let trait_id = db.well_known_trait_id(well_known).unwrap();
+    let trait_id = match db.well_known_trait_id(well_known) {
+        Some(id) => id,
+        None => return Ok(()),
+    };
```

Comment:
Why this change? I think it makes more sense to error early here if we misuse this function.